### PR TITLE
[Native clouds] modularization

### PR DESCRIPTION
**Pull request by istraka**
Introduce SPI api and break functionality into modules.

FYI @mchoma

### CODE REVIEW

File: clouds/clouds-azure/src/main/java/azure/core/AzureSunstoneDeployerProvider.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import sunstone.core.api.SunstoneCloudDeployer;
+import sunstone.core.spi.SunstoneCloudDeployerProvider;
+
+public class AzureSunstoneDeployerProvider implements SunstoneCloudDeployerProvider {
+    @Override
+    public SunstoneCloudDeployer create(Class annotation) {
+        if (WithAzureArmTemplate.class.isAssignableFrom(annotation)) {
+            return new AzureSunstoneDeployer();
+        }
+        return null;
```
Comment (mnovak1) [inline clouds/clouds-azure/src/main/java/azure/core/AzureSunstoneDeployerProvider.java:14]: `Optional` could be used here.

File: clouds/clouds-aws/src/main/java/aws/core/AwsSunstoneStore.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+/**
+ * An abstraction over JUnit5 Extension store providing methods to set commonly used resources.
+ */
+class AwsSunstoneStore extends SunstoneStore {
+    private static final ExtensionContext.Namespace NAMESPACE = ExtensionContext.Namespace.create("sunstone", "core", "SunstoneStore");
+    private static final String CLOSABLES = "closables";
+
+
+    private static final String SUITE_LEVEL_DEPLOYMENTS = "sunstoneSuiteLevelDeployments";
```
[the remark this answers is not in the document]
Comment (istraka) [reply clouds/clouds-aws/src/main/java/aws/core/AwsSunstoneStore.java:19]: Yep, I have forgot to clean it properly.

File: clouds/clouds-aws/src/main/java/aws/core/AwsSunstoneDeployerProvider.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import sunstone.core.api.SunstoneCloudDeployer;
+import sunstone.core.spi.SunstoneCloudDeployerProvider;
+
+public class AwsSunstoneDeployerProvider implements SunstoneCloudDeployerProvider {
+    @Override
+    public SunstoneCloudDeployer create(Class annotation) {
+        if (WithAwsCfTemplate.class.isAssignableFrom(annotation)) {
+            return new AwsSunstoneDeployer();
+        }
+        return null;
```
[the remark this answers is not in the document]
Comment (istraka) [reply clouds/clouds-aws/src/main/java/aws/core/AwsSunstoneDeployerProvider.java]: Done.

File: clouds/clouds-aws/src/main/java/aws/core/AwsSunstoneStore.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (istraka) [reply clouds/clouds-aws/src/main/java/aws/core/AwsSunstoneStore.java]: Done.

File: clouds/clouds-azure/src/main/java/azure/core/AzureSunstoneDeployerProvider.java (hunk 1 above)
Comment (istraka) [reply clouds/clouds-azure/src/main/java/azure/core/AzureSunstoneDeployerProvider.java]: Done.

**Comment by istraka** [comment]
@mnovak1 I have fixed issues. Can you kindly review again?

**Comment by mnovak1** [comment]
LGTM

**Review by mnovak1 — APPROVED** [review]

File: clouds/clouds-core/src/main/java/sunstone/api/WithAwsCfTemplate.java [hunk 4]
```diff
@@ -33,7 +33,7 @@
      * Region that should be used for creating resource group. Expression is allowed, e.g. {@code abc-${var:default}-xyz} -
```
Comment (mchoma) [inline clouds/clouds-core/src/main/java/sunstone/api/WithAwsCfTemplate.java:33]: `WithAwsCfTemplate`

To be honest I wouldn't expect keywords `Aws`  or `Azure` in core module.

**Pull request merged**